File: web/src/app/admin/teacher/assessment.ts
import { AssessmentStatus } from './assessment-status';
import { AssessmentType } from './assessment-type';

export interface Assessment {
  id?: number | null;
  createdAt?: string | null;
  updatedAt?: string | null;
  schoolId?: number | null;
  schoolName?: string | null;
  subjectAssignmentId?: number | null;
  subjectId?: number | null;
  subjectCode?: string | null;
  subjectName?: string | null;
  gradeId?: number | null;
  gradeName?: string | null;
  teacherId?: number | null;
  teacherName?: string | null;
  title?: string | null;
  description?: string | null;
  type?: AssessmentType | null;
  status?: AssessmentStatus | null;
  assessmentDate?: string | null;
  totalMarks?: number | null;
  passMark?: number | null;
  studentCount?: number | null;
  markedCount?: number | null;
  averageScore?: number | null;
  averagePercentage?: number | null;
}

File: api/src/main/java/com/tsoinyane/api/assessment/Assessment.java
package com.tsoinyane.api.assessment;

import com.tsoinyane.api.common.BaseEntity;
import com.tsoinyane.api.subjectassignment.SubjectAssignment;
import com.tsoinyane.api.teacher.Teacher;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@Entity
@Table(name = "assessment")
@EqualsAndHashCode(callSuper = true, exclude = {"subjectAssignment", "teacher"})
@ToString(callSuper = true, exclude = {"subjectAssignment", "teacher"})
public class Assessment extends BaseEntity {

    @Column(nullable = false, length = 150)
    private String title;

    @Column(length = 1500)
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 30)
    private AssessmentType type;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 30)
    private AssessmentStatus status;

    @Column(name = "assessment_date")
    private LocalDateTime assessmentDate;

    @Column(name = "total_marks", nullable = false)
    private Double totalMarks;

    @Column(name = "pass_mark")
    private Double passMark;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "subject_assignment_id", nullable = false)
    private SubjectAssignment subjectAssignment;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "teacher_id", nullable = false)
    private Teacher teacher;
}
